services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java

```diff
@@ -84,23 +80,25 @@ public ReadRequestThrottler(
   public ReadRequestThrottler(
       ZkRoutersClusterManager zkRoutersManager,
       ReadOnlyStoreRepository storeRepository,
-      RoutingDataRepository routingDataRepository,
       long maxRouterReadCapacity,
       AggRouterHttpRequestStats stats,
       double perStoreRouterQuotaBuffer,
       long storeQuotaCheckTimeWindow) {
     this.zkRoutersManager = zkRoutersManager;
     this.storeRepository = storeRepository;
-    this.routingDataRepository = routingDataRepository;
     this.storeQuotaCheckTimeWindow = storeQuotaCheckTimeWindow;
-    this.storeRepository.registerStoreDataChangedListener(this);
     this.stats = stats;
     this.maxRouterReadCapacity = maxRouterReadCapacity;
     this.lastRouterCount = zkRoutersManager.getExpectedRoutersCount();
     this.perStoreRouterQuotaBuffer = perStoreRouterQuotaBuffer;
     this.idealTotalQuotaPerRouter = calculateIdealTotalQuotaPerRouter();
-    this.storesThrottlers = new AtomicReference<>(buildAllStoreReadThrottlers());
     this.isNoopThrottlerEnabled = false;
+
+    /** Calling {@link #buildAllStoreReadThrottlers()} should be done after all internal state is initialized */
```

Comment:
Looks like there was another bug in the initialization in the constructor of this class. https://github.com/linkedin/venice/pull/399/files

Thanks for catching another!

Reply:
Yep, this class is the gift that keeps on giving 🤣